app/src/components/token/Token.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { HTMLProps } from "react";
+import { css } from "@emotion/react";
+
+import { Icon, Icons } from "@phoenix/components";
+import { SizingProps, StylableProps } from "@phoenix/components/types";
+import { useTheme } from "@phoenix/contexts";
+
+interface TokenProps
+  extends Omit<HTMLProps<HTMLDivElement>, "size" | "css">,
+    StylableProps,
+    SizingProps {
+  children?: React.ReactNode;
+  /**
+   * Leading visual element
+   */
+  leadingVisual?: React.ReactNode;
+  /**
+   * Whether the token is disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * The color of the token.
+   *
+   * Can be any valid CSS color value, including CSS variables.
+   */
+  color?: string;
+  /**
+   * The function to call when the token is pressed
+   */
+  onPress?: () => void;
+  /**
+   * The function to call when the token is removed.
+   *
+   * If provided, an icon button will be displayed to the right of the token.
+   */
+  onRemove?: () => void;
+}
+
+const tokenBaseCSS = css`
+  display: inline-flex;
+  align-items: center;
+  gap: var(--ac-global-dimension-static-size-100);
+  font-size: var(--ac-global-dimension-static-font-size-75);
+  line-height: var(--ac-global-line-height-s);
+  padding: 0 var(--ac-global-dimension-static-size-100);
+  border-radius: var(--ac-global-rounding-large);
+  border: 1px solid
+    lch(from var(--ac-internal-token-color) calc((l) * infinity) c h / 0.3);
+  color: lch(from var(--ac-internal-token-color) calc((50 - l) * infinity) 0 0);
+  user-select: none;
+
+  &[data-size="S"] {
+    height: var(--ac-global-dimension-static-size-200);
+  }
+
+  &[data-size="M"] {
+    height: var(--ac-global-dimension-static-size-250);
+  }
+
+  &[data-size="L"] {
+    height: var(--ac-global-dimension-static-size-300);
+  }
+
+  &[data-disabled] {
+    opacity: 0.5;
+    cursor: not-allowed;
+  }
+
+  &[data-theme="light"] {
+    background: var(--ac-internal-token-color);
+  }
+
+  &[data-theme="dark"] {
+    // generate a new dark token bg color from the input color
+    --px-internal-dark-token-bg: lch(
```

Comment:
what's the internal meant to signify - that it's scoped to the component? Might not be necessary?

Reply:
Yeah it's a variable meant to be scoped to the component. Wasn't really sure how to signify this. Just want to avoid future name collisions

Reply:
Renamed a bit